lib/puppet/type/rds_instance.rb

```diff
@@ -0,0 +1,184 @@
+Puppet::Type.newtype(:rds_instance) do
+  @doc = 'type representing an RDS instance'
+
+  ensurable do
+    defaultvalues
+    aliasvalue(:available, :present)
+    newvalue(:destroy) do
+      provider.destroy
+    end
+  end
+
+  newparam(:name, namevar: true) do
+    desc 'the name of the db instance (also known as the db_instance_identifier)'
+    validate do |value|
+    end
+  end
+
+  newparam(:db_name) do
```

Comment:
Any reason for this being a param rather than a property? I see you populate the property_hash with the value and it feels useful to puppet resource?